src/complexity_measures/sample_entropy.jl

```diff
@@ -0,0 +1,144 @@
+using DelayEmbeddings
+using Statistics
+using Neighborhood.NearestNeighbors: inrangecount, Chebyshev
+
+export SampleEntropy
+
+"""
+    SampleEntropy(; r::Real = 0.2 * Statistics.std(x), m::Int = 2, τ::Int = 1,
+        metric = Chebyshev()
```

Comment:
The keywords in the docstring are a lot. Can we have a `## Keyword Arguments` section instead? Furthermore, the keyword `r::Real = 0.2 * Statistics.std(x)` doesn't make sense because `x` isn't given in that call signature. I propose:

```
SampleEntropy([x]; r = 0.2std(x), kwargs...)
```
and say that if `x` is not given, keyword `r` is mandatory. I don't agree with `r` defaulting to `0.1` if `x` is not given, this doesn't make sense. Unlike `metric, τ,` the numeric value of `r` depends entirely on the scale of the data `x`.

(also, there is no reason to say that `std` comes from `Statistics`, 99.9% of our users will probably know `std` before they come across our package)

Reply:
> and say that if x is not given, keyword r is mandatory. I don't agree with r defaulting to 0.1 if x is not given, this doesn't make sense. Unlike metric, τ, the numeric value of r depends entirely on the scale of the data x.

Sure, that makes sense!

Reply:
> (also, there is no reason to say that std comes from Statistics, 99.9% of our users will probably know std before they come across our package)

... and that also makes sense!